### PR TITLE
Observer pattern event tests using spies

### PR DESCRIPTION
**Pull request by KaseyCantu**

### CODE REVIEW

File: tests/events/test_request_sent_event.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from pytest_mock import MockerFixture
+
+# from shipengine_sdk.events import ShipEngineEventListener
+
+# from ..util import stub_shipengine_config
+
+
+class TestRequestSentEvent:
+    def test_config_with_retries_disabled(self, mocker: MockerFixture) -> None:
+        """Tests that the SDK does not automatically retry if retries in config is set to 0."""
```
Comment (pierce-h) [inline tests/events/test_request_sent_event.py:11]: Is this spec running?

**Review by pierce-h — APPROVED** [review]
@KaseyCantu these changes seem reasonable. I wish I knew python better so I could give you a better review.

File: README.md [hunk 2]
```diff
@@ -89,7 +89,15 @@ poetry install
...
  ```
 
-### Pre-Commit Hooks
+## Adding dependencies to the project
+If your changes require you to install a python package/module using `poetry add <some package>` or
+`poetry add <some package> -D` for a dev dependency. You will also need to run the following command to
+regenerate a `requirements.txt` file that includes the newly added dependencies:
+```bash
+poetry export -f requirements.txt --output requirements.txt --without-hashes --dev
+```
```
Comment (nate-williams-auctane) [inline README.md:98]: 👍

**Review by nate-williams-auctane — APPROVED** [review]
Looks good, I'm no expert on the mocks, but the packages and the approach look good to this point!

**Pull request merged**